crates/fuel-core/src/state/in_memory/memory_store.rs

```diff
@@ -4,35 +4,28 @@ use crate::{
         Result as DatabaseResult,
     },
     state::{
-        in_memory::{
-            column_key,
-            is_column,
-        },
         BatchOperations,
-        ColumnId,
         IterDirection,
         KVItem,
         KeyValueStore,
         TransactableStorage,
+        Value,
     },
 };
 use fuel_core_storage::iter::{
     BoxedIter,
     IntoBoxedIter,
 };
-use itertools::Itertools;
 use std::{
-    collections::HashMap,
+    collections::BTreeMap,
     fmt::Debug,
-    mem::size_of,
     sync::Mutex,
 };
 
 #[derive(Default, Debug)]
 pub struct MemoryStore {
-    // TODO: Remove `Mutex` and usage of the `column_key`.
-    // TODO: Use `BTreeMap`.
-    inner: Mutex<HashMap<Vec<u8>, Vec<u8>>>,
+    // TODO: Remove `Mutex`.
+    inner: [Mutex<BTreeMap<Vec<u8>, Value>>; Column::COUNT],
```

Comment:
Can't this Mutex just be removed? This type is not clone so it's not being shared anyway

Reply:
Because it requires modification of the `KeyValueStore` trait to use `&mut self`, but it requires a huge refactoring